conformance/lang/expressions/multiplicative-expr/division.balt

```diff
@@ -4033,3 +4033,922 @@ Description: Test integer division panic at module-level.
+Description: Test float division with optional integers.
+Labels: DecimalNumber, float, float-NaN, int, multiplicative-expr, nil-literal, numeric-nil-lifting
+
+function init() {
+    int? a = 4;
+    int? b = ();
+    float c = 4.5e-1;
+    float d = float:NaN;
+
+    float? v = c / a;
+    io:println(v); // @output 0.1125
+
+    v = c / b;
+    io:println(v); // @output ()
+
+    v = d / a;
+    io:println(v); // @output NaN
+
+    v = d / b;
+    io:println(v); // @output ()
+}
+
+Test-Case: output
+Description: Test optional float division with optional and non-optional integers.
+Labels: DecimalNumber, float, float-NaN, int, multiplicative-expr, nil-literal, numeric-nil-lifting
+
+function init() {
+    int a = 2; 
+    int? b = 4;
+    int? c = ();
+    float? d = 4.5e-1;
+    float? e = float:NaN;
+
+    float? v = d / a;
+    io:println(v); // @output 0.225
+
+    v = e / a;
+    io:println(v); // @output NaN
+
+    v = d / b;
+    io:println(v); // @output 0.1125
+
+    v = e / b;
+    io:println(v); // @output NaN
+
+    v = d / c;
+    io:println(v); // @output ()
+
+    v = e / c;
+    io:println(v); // @output ()
+}
+
+Test-Case: output
+Description: Test float division with optional subtypes of integers.
+Labels: DecimalNumber, float, float-NaN, int, multiplicative-expr, numeric-nil-lifting
+
+type Ints -9|9;
+
+function init() {
+    int:Signed8? a = -2;
+    int:Signed16? b = 2;
+    int:Signed32? c = 3;
+    int:Unsigned8? d = 4;
+    int:Unsigned16? e = 5;
+    int:Unsigned32? f = 10;
+    byte? g = 2;
+    Ints? h = 9;
+
+    float i = 2.5;
+
+    float? v = i / a;
+    io:println(v); // @output -1.25
+
+    v = i / b;
+    io:println(v); // @output 1.25
+
+    v = i / c;
+    io:println(v); // @output 0.8333333333333334
+
+    v = i / d;
+    io:println(v); // @output 0.625
+
+    v = i / e;
+    io:println(v); // @output 0.5
+
+    v = i / f;
+    io:println(v); // @output 0.25
+
+    v = i / g;
+    io:println(v); // @output 1.25
+
+    v = i / h;
+    io:println(v); // @output 0.2777777777777778
+}
+
+Test-Case: output
+Description: Test float division with optional subtypes of integers initialized with nil-literal.
+Labels: DecimalNumber, float, int, multiplicative-expr, nil-literal, numeric-nil-lifting
+
+function init() {
+    int:Signed8? a = ();
+    int:Signed16? b = ();
+    int:Signed32? c = ();
+    int:Unsigned8? d = ();
+    int:Unsigned16? e = ();
+    int:Unsigned32? f = ();
+    byte? g = ();
+
+    float h = 2.5;
+
+    float? v = h / a;
+    io:println(v); // @output ()
+
+    v = h / b;
+    io:println(v); // @output ()
+
+    v = h / c;
+    io:println(v); // @output ()
+
+    v = h / d;
+    io:println(v); // @output ()
+
+    v = h / e;
+    io:println(v); // @output ()
+
+    v = h / f;
+    io:println(v); // @output ()
+
+    v = h / g;
+    io:println(v); // @output ()
+}
+
+Test-Case: output
+Description: Test optional float division with optional and non-optional subtypes of integers.
```

Comment:
Let's split the optional and non optional ones into two tests.

Reply:
For each of these tests with optional types, there are four different possible ways of doing the calculation. Hence let's add all possible scenarios separately with each of these types.